File: src/main/java/burrow/furniture/bincore/ShellNewCommand.java
package burrow.furniture.bincore;

import burrow.core.command.Command;
import burrow.core.command.CommandContext;
import burrow.core.command.CommandType;
import org.jetbrains.annotations.NotNull;
import picocli.CommandLine;

import java.io.IOException;

@CommandLine.Command(name = "shell.new", description = "Create a shell file.")
@CommandType(BinCoreFurniture.COMMAND_TYPE)
public class ShellNewCommand extends Command {
    @CommandLine.Option(
        names = {"--force", "-f"},
        paramLabel = "is-force",
        description = "Create a shell file, whether or not it already exists.",
        defaultValue = "false"
    )
    public Boolean isForce;

    public ShellNewCommand(@NotNull final CommandContext commandContext) {
        super(commandContext);
    }

    @Override
    public Integer call() throws IOException {
        final var binCoreFurniture = use(BinCoreFurniture.class);
        final var shellContent = binCoreFurniture.getDefaultShellContent();
        if (isForce) {
            binCoreFurniture.createShellFile(shellContent);
        } else {
            final var createdFile = binCoreFurniture.createShellFileIfAbsent(shellContent);
            if (!createdFile) {
                buffer.append("Shell file is not created because it already exists. Use --force option to create a new shell file.");
            }
        }

        return CommandLine.ExitCode.OK;
    }
}

File: src/main/kotlin/carton/shell/command/ShellNewCommand.kt
package burrow.carton.shell.command

import burrow.carton.shell.Shell
import burrow.kernel.terminal.*
import kotlin.io.path.exists

@BurrowCommand(
    name = "shell.new",
    header = ["Creates a new shell file."]
)
class ShellNewCommand(data: CommandData) : Command(data) {
    @Option(
        names = ["--force", "-f"],
        paramLabel = "force",
        description = ["Create a shell file no matter if it already exists."]
    )
    var shouldBeForce: Boolean = false

    override fun call(): Int {
        val bin = use(Shell::class)
        val binFile = bin.getShellFilePath()
        val shellContent = bin.getDefaultShellContent()
        if (shouldBeForce) {
            bin.createShellFile(shellContent)
            stdout.println("Created shell file: $binFile")
        } else {
            if (binFile.exists()) {
                stderr.println("Shell file already exists: $binFile")
                return ExitCode.USAGE
            }
            bin.createShellFile(shellContent)
        }

        return ExitCode.OK
    }
}